src/structures/paging/mapper/mapped_page_table.rs

```diff
@@ -140,6 +141,418 @@ impl<'a, P: PageTableFrameMapping> MappedPageTable<'a, P> {
 
         Ok(MapperFlush::new(page))
     }
+
+    #[inline]
+    fn next_table_fn_create_next_table<'b, A>(
+        (flags, allocator): &mut (PageTableFlags, &mut A),
+        entry: &'b mut PageTableEntry,
+        walker: &PageTableWalker<P>,
+    ) -> Result<&'b mut PageTable, PageTableCreateError>
+    where
+        A: FrameAllocator<Size4KiB> + ?Sized,
+    {
+        walker
+            .create_next_table(entry, *flags, *allocator)
+            .map_err(Into::into)
+    }
+
+    #[inline]
+    fn next_table_fn_next_table_mut<'b, T>(
+        _: &mut T,
+        entry: &'b mut PageTableEntry,
+        walker: &PageTableWalker<P>,
+    ) -> Result<&'b mut PageTable, PageTableWalkError> {
+        walker.next_table_mut(entry)
+    }
+
+    fn modify_range_1gib<ModifyFn, ModifyInfo, Err, NextTableFn, NextTableFnErr>(
+        &mut self,
+        pages: PageRange<Size1GiB>,
+        modify: ModifyFn,
+        mut info: ModifyInfo,
+        next_table: NextTableFn,
+    ) -> Result<MapperFlushRange<Size1GiB>, (Err, MapperFlushRange<Size1GiB>)>
+    where
+        ModifyFn: Fn(&mut PageTableEntry, Page<Size1GiB>, &mut ModifyInfo) -> Result<(), Err>,
+        NextTableFn: for<'b> Fn(
+            &mut ModifyInfo,
+            &'b mut PageTableEntry,
+            &PageTableWalker<P>,
+        ) -> Result<&'b mut PageTable, NextTableFnErr>,
+        NextTableFnErr: Into<Err>,
+    {
+        if pages.is_empty() {
+            return Ok(MapperFlushRange::empty());
+        }
+
+        let p4 = &mut self.level_4_table;
+        let page_table_walker = &mut self.page_table_walker;
+
+        (pages.start.p4_index().into()..=pages.end.p4_index().into())
```

Comment:
Why are you using an inclusive range here, even though the `pages` argument is an exclusive range?

Reply:
I used an inclusive range because I'm using the range between the p4_indices not the pages. For example, for a range starting with `0x8000000000` and ending `0x8040000000` both addresses share the same `p4_index`.

Reply:
Makes sense, thanks for the explanation!